modules/util/portal-tools-db-partition-virtual-instance-migration-extractor/src/main/java/com/liferay/portal/tools/db/partition/virtual/instance/migration/extractor/DBPartitionVirtualInstanceMigrationExtractor.java

```diff
@@ -0,0 +1,201 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.tools.db.partition.virtual.instance.migration.extractor;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.fasterxml.jackson.databind.util.ISO8601DateFormat;
+
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.tools.db.partition.virtual.instance.migration.common.InstanceData;
+import com.liferay.portal.tools.db.partition.virtual.instance.migration.extractor.util.DatabaseUtil;
+
+import java.io.File;
+
+import java.nio.file.Files;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import java.text.SimpleDateFormat;
+
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.CommandLineParser;
+import org.apache.commons.cli.DefaultParser;
+import org.apache.commons.cli.HelpFormatter;
+import org.apache.commons.cli.Options;
+import org.apache.commons.cli.ParseException;
+
+/**
+ * @author Luis Ortiz
+ */
+public class DBPartitionVirtualInstanceMigrationExtractor {
+
+	public static void main(String[] args) {
+		try {
+			_main(args);
+		}
+		catch (Exception exception) {
+			System.err.println("Unexpected error:");
+
+			exception.printStackTrace();
+
+			_exit(_LIFERAY_COMMON_EXIT_CODE_BAD);
+		}
+	}
+
+	private static void _exit(int code) {
+		try {
+			if (_connection != null) {
+				_connection.close();
+			}
+		}
+		catch (SQLException sqlException) {
+			System.err.println(sqlException);
+		}
+
+		if (code != _LIFERAY_COMMON_EXIT_CODE_OK) {
+			System.exit(code);
+		}
+	}
+
+	private static Options _getOptions() {
+		Options options = new Options();
+
+		options.addOption("h", "help", false, "Print help message.");
+		options.addRequiredOption("url", "jdbc-url", true, "Set the JDBC URL.");
+		options.addRequiredOption(
+			"pass", "password", true, "Set the password.");
+		options.addRequiredOption("user", "user", true, "Set the user.");
+		options.addRequiredOption(
```

Comment:
Please mark the partition as optional since, in case a partition is not provided, we must use the default. it is more intuitive for LXC to extract the target.